src/Core/AbstractTest.php

```diff
@@ -182,6 +182,30 @@ public function setPending($state)
     }
 
     /**
+     * Set the focused status of the test and its children according to the
+     * supplied focus pattern and/or skip pattern
+     *
+     * @param string|null $focusPattern
+     * @param string|null $skipPattern
+     */
+    public function applyFocusPatterns($focusPattern, $skipPattern = null)
+    {
+        $title = $this->getTitle();
+
+        if ($skipPattern !== null && preg_match($skipPattern, $title)) {
```

Comment:
We could argue that an invalid regex is the tester's problem, but a more helpful error message might be nice